File: 2024/matlab/aoc1.m
input_path = fullfile(get_input_root,'day_1.txt');

x = read_txt(input_path);
x = strip(x);
lines = strsplit(x,crlf);

%part1 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
x = [];
for i1 = 1:numel(lines)
    v = strsplit(lines{i1},' ');
    x = cat(1,x,str2double(v));
end

L1 = sort(x(:,1));
L2 = sort(x(:,2));

disp(sum(abs(L1 - L2)))

%part 2 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
s = 0;
for i1 = 1:size(x,1)
   si1 = sum(x(i1,1) == x(:,2))*x(i1,1);
   s = s + si1; 
end

disp(s)

File: 2024/python/aoc1.py
# -*- coding: utf-8 -*-
import numpy as np

input_path = '..\inputs\day_1.txt'
txt = open(input_path).read()

lines = txt.split('\n')

v = [x.split('  ') for x in lines]
L = list(zip(*v))

#part 1 #######################################################################

L[0] = np.array([int(x) for x in L[0]])
L[1] = np.array([int(x) for x in L[1]])

L[0].sort()
L[1].sort()

print(sum(abs((L[0]) - L[1])))

#part 2 #######################################################################
s = 0
for i in range(0,len(L[0])):
    s += sum(L[0][i] == L[1])*L[0][i]

print(s)